src/Handler/Workspace/Create/CreateWorkspaceHandler.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery\Handler\Workspace\Create;
+
+use Exception;
+use Google\Protobuf\Internal\Message;
+use Google_Service_CloudResourceManager_Binding;
+use Google_Service_CloudResourceManager_GetIamPolicyRequest;
+use Google_Service_CloudResourceManager_Policy;
+use Google_Service_CloudResourceManager_SetIamPolicyRequest;
+use Google_Service_Iam_CreateServiceAccountKeyRequest;
+use Google_Service_Iam_CreateServiceAccountRequest;
+use Keboola\StorageDriver\BigQuery\CredentialsHelper;
+use Keboola\StorageDriver\BigQuery\GCPClientManager;
+use Keboola\StorageDriver\BigQuery\IAmPermissions;
+use Keboola\StorageDriver\BigQuery\NameGenerator;
+use Keboola\StorageDriver\Command\Workspace\CreateWorkspaceCommand;
+use Keboola\StorageDriver\Command\Workspace\CreateWorkspaceResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+
+final class CreateWorkspaceHandler implements DriverCommandHandlerInterface
+{
+    public const PRIVATE_KEY_TYPE = 'TYPE_GOOGLE_CREDENTIALS_FILE';
+    public const KEY_DATA_PROPERTY_PRIVATE_KEY = 'private_key';
+
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     * @param CreateWorkspaceCommand $command
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof CreateWorkspaceCommand);
+
+        // validate
+        assert($command->getStackPrefix() !== '', 'CreateWorkspaceCommand.stackPrefix is required');
+        assert($command->getWorkspaceId() !== '', 'CreateWorkspaceCommand.workspaceId is required');
+        assert(
+            $command->getProjectReadOnlyRoleName() !== '',
+            'CreateWorkspaceCommand.projectReadOnlyRoleName is required',
+        );
+
+        $bqClient = $this->clientManager->getBigQueryClient($credentials);
+        $projectCredentials = CredentialsHelper::getCredentialsArray($credentials);
+
+        $nameGenerator = new NameGenerator($command->getStackPrefix());
+        $newWsDatasetName = $nameGenerator->createWorkspaceObjectNameForWorkspaceId($command->getWorkspaceId());
+        $newWsServiceAccName = $nameGenerator->createWorkspaceUserNameForWorkspaceId($command->getWorkspaceId());
+
+        // create WS service acc
+        $iamService = $this->clientManager->getIamClient($credentials);
+        $serviceAccountsService = $iamService->projects_serviceAccounts;
+        $createServiceAccountRequest = new Google_Service_Iam_CreateServiceAccountRequest();
```

Comment:
Tady všecky ty googlí třidy mají PSR4 názvy aj, imho bysme měli používat ty a ne PSR0
`\Google\Service\Iam\CreateServiceAccountRequest`

Reply:
Jo super dik upravim to tu potom este upravim vsade v kode